### ISSUE TITLE
nimble-dialog has accessibility violations

### ISSUE THREAD

**Issue by mollykreis** [ni/nimble#689]
# 🐛 Bug Report

The `nimble-dialog` has accessibility violations associated with it in storybook.

The violation is that `aria-label` is not a valid attribute to set on a `nimble-dialog` element because the element doesn't have a role associated with it. I believe the solution is to mirror setting `role=alertdialog` on the dialog's `<template>`, like it is set on in inner `<dialog>`.

## 🤔 Expected Behavior

We shouldn't have accessibility violations on our components.

## 😯 Current Behavior

We have an accessibility violation on our component.

## 💁 Possible Solution

Set `role="alertdialog"` on the nimble-dialog's `<template>` element.

## 🌍 Your Environment

* OS & Device: [e.g. MacOS, iOS, Windows, Linux] on [iPhone 7, PC]
* Browser [e.g. Microsoft Edge, Google Chrome, Apple Safari, Mozilla FireFox]
* Version [e.g. 1.8.0]
